pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -68,8 +94,10 @@
 
   private PinotDataBuffer headerBuffer;
   private List<PinotDataBuffer> dataBuffers = new ArrayList<>();
-  private FixedByteSingleValueMultiColWriter headerWriter;
-  private FixedByteSingleValueMultiColReader headerReader;
+  private List<PinotDataBuffer> headerBuffers = new ArrayList<>();
```

Comment:
As yet, the number of buffers is 1, because we still limit by the number of rows.

When we start limiting by memory (once we have the other pieces of the puzzle fixed to use off-heap) we can work on tuning this to coalesce if necessary. The idea is to try to keep this to 1 or a few at best, by taking hints from the previous completed segment